src/test/java/redis/clients/jedis/tests/JedisPoolTest.java

```diff
@@ -381,4 +382,41 @@ private int getClientCount(final String clientList) {
     return clientList.split("\n").length;
   }
 
+  @Test(expected = JedisConnectionException.class)
+  public void testResetInvalidPassword() {
+    JedisFactory factory = new JedisFactory(hnp.getHost(), hnp.getPort(), 2000, 2000, "foobared", 0, "my_shiny_client_name", false, null, null, null);
+    JedisPool pool = new JedisPool(new JedisPoolConfig(), factory);
+    Jedis obj1 = pool.getResource();
+    obj1.set("foo", "bar");
+    assertEquals("bar", obj1.get("foo"));
+    assertEquals(1, pool.getNumActive());
+    
```

Comment:
Move `obj1.close()` here. There's nothing related to `obj1` in try.

Reply:
done

Reply:
@sazzad16 if I move obj1.close() here, after calling factory.setPassword("wrong password"), the pool.getResource() will not make a new object for Jedis. The old Jedis which has the correct password got returned.

Reply:
@sazzad16 I changed to use Jedis.quit and it works well.